firstvoices/backend/search/indices/dictionary_entry_document.py

```diff
@@ -207,38 +192,39 @@ def update_notes(sender, instance, **kwargs):
         )
 
 
-# If a site is deleted, delete all docs from index related to site
-@receiver(post_delete, sender=Site)
-def delete_related_docs(sender, instance, **kwargs):
+# Acknowledgement update
+@receiver(post_delete, sender=Acknowledgement)
+@receiver(post_save, sender=Acknowledgement)
+def update_acknowledgement(sender, instance, **kwargs):
     logger = logging.getLogger(ELASTICSEARCH_LOGGER)
-    dictionary_entries_set = instance.dictionaryentry_set.all()
+    dictionary_entry = instance.dictionary_entry
+    acknowledgements_text = get_acknowledgements_text(dictionary_entry)
 
-    for dictionary_entry in dictionary_entries_set:
-        try:
-            existing_entry = get_object_from_index(
-                ELASTICSEARCH_DICTIONARY_ENTRY_INDEX, dictionary_entry.id
-            )
-            if not existing_entry:
-                raise NotFoundError
+    try:
+        existing_entry = get_object_from_index(
+            ELASTICSEARCH_DICTIONARY_ENTRY_INDEX, dictionary_entry.id
```

Comment:
Minor: I'm pretty sure you can get this id from the acknowledgement instance without doing a separate lookup for the dictionary entry. `instance.dictionary_entry_id` should be available because it is the foreign key value

Reply:
Yes, you're right. we can get the dictionary_entry id from the acknowledgement instance. But, this is to get the elastic search document id that we need to fetch the elasticsearch doc to update/delete it in the index for the respective operation.